src/OpenTelemetry.AutoInstrumentation/Instrumentations/MongoDb/MongoClientIntegration.cs

```diff
@@ -0,0 +1,123 @@
+// <copyright file="MongoClientIntegration.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using OpenTelemetry.AutoInstrumentation.CallTarget;
+
+namespace OpenTelemetry.AutoInstrumentation.Instrumentations.MongoDb
+{
+    /// <summary>
+    /// MongoDB.Driver.MongoClient calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "MongoDB.Driver",
+        TypeName = "MongoDB.Driver.MongoClient",
+        MethodName = ".ctor",
+        ReturnTypeName = ClrNames.Void,
+        ParameterTypeNames = new[] { "MongoDB.Driver.MongoClientSettings" },
+        MinimumVersion = "2.3.0",
+        MaximumVersion = "2.65535.65535",
+        IntegrationName = "MongoDb")]
+    public class MongoClientIntegration
+    {
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TMongoClientSettings">Type of the settings</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="settings">The source of the original GraphQL query</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget, TMongoClientSettings>(TTarget instance, TMongoClientSettings settings)
+        {
+            // Load driver adapter
+            Type driverAcitivitySubscriber = Type.GetType("MongoDB.Driver.Core.Extensions.DiagnosticSources.DiagnosticsActivityEventSubscriber, MongoDB.Driver.Core.Extensions.DiagnosticSources");
+            if (driverAcitivitySubscriber == null)
+            {
+                string basePath = Environment.GetEnvironmentVariable("DOTNET_SHARED_STORE");
+                string processArch = Environment.Is64BitProcess ? "x64" : "x32";
+                string assemblyPath = Path.Combine(basePath, processArch, @"netcoreapp3.1\mongodb.driver.core.extensions.diagnosticsources\1.1.0\lib\netstandard2.0\MongoDB.Driver.Core.Extensions.DiagnosticSources.dll");
```

Comment:
Any ways to prob this path automatically? So we can only use Assembly.Load?

Reply:
@rajkumar-rangaraj thanks for the AdditionalDeps clue. I completely forgot that it works in pair with shared store and actually defines what and where to load. Managed to cleanup all that loading.

Reply:
... btw, do you also know if we can only include `MongoDB.Driver.Core.Extensions.DiagnosticSources.dll` in the shared store? Currently it's pulling all of the sub dependencies also 🤔

Reply:
@rajkumar-rangaraj any ideas ⬆️ ? It will help to optimize delivery by removing 2 unnecessary packages.